### DISCUSSION TITLE
[Question] Will this package continue to work with Twitter's ban on third-party clients?

### DISCUSSION THREAD

**Discussion by jamesgeer** [plhery/node-twitter-api-v2#444] [General]
Just Googling "twitter api" and there are a lot of news articles about certain applications being suspended with Twitter changing their terms and conditions, will this mean the Twitter API is being terminated, and thus this package will stop working?

I'm working on a University project that relies on this package (and the Twitter API) so I'm uncertain whether to stick with it for now (the project is due in 10 weeks) or make the effort to switch to another platform such as LinkedIn.

TL;DR: Will this library continue to work for posting tweets?

**Comment by plhery** [comment]
Hello,

I'm not working at/for Twitter, also sad about what happened to Twitter ; and can't predict the future as they make surprise moves.

For now, they clarified that only 3rd party clients were banned, so other uses are still possible, and this package will continue to work as long as they provide their API v1 or v2.

Best,

Paul
